src/builder.rs

```diff
@@ -16,7 +16,7 @@ use crate::{
 
 /// Builder for Service
 pub struct ServiceBuilder {
-    inner: HashMap<String, ProtocolMeta>,
+    inner: HashMap<ProtocolId, ProtocolMeta>,
```

Comment:
What’s the cause that the key is String before, and why it is safe to use `ProtocolId` instead?

Reply:
In a lot of our scenarios, we look up the corresponding configuration based on id, and the original name-based lookup leads to various O(n) lookups